frontend/webservice/vrf.cgi

```diff
@@ -790,6 +842,18 @@ sub provision_vrf{
     }
     my $pending_vrf = $vrf->to_hash;
 
+    my $error = OESS::DB::VRF::add_vrf_history(
+        db => $db,
+        event => $type eq 'provisioned' ? 'create' : 'edit',
```

Comment:
will $type ever be anything but 'provisioned'?

Reply:
When a vrf is edited, it calls the same provision_vrf method, $type is used to determine if it is provisioned or edited. The other option is 'modified'. So if the vrf is not being 'provisioned' it is being 'modified'.

Reply:
But I don't see where `$type` is ever set to 'modified'.

Reply:
I see, the add_vrf_history used to be at the bottom of the provision_vrf method because there is an if statement that checks if the vrf_id already exists. Line 893 is where $type can be set to 'modified' when editing a vrf. This conflicts with having the add_vrf_history method moved above any $db->commit.

Reply:
I just made two separate method calls to add_vrf_history and set the events